docs/docs/sourcing-from-kentico-cloud.md

```diff
@@ -120,41 +120,50 @@ exports.onCreateNode = ({ node, actions: { createNodeField } }) => {
 Now that you have a pretty way to define the path for your pages, you can create the pages programmatically:
 
 ```javascript:title=gatsby-node.js
-const path = require(`path`);
-
-exports.onCreateNode ... // As above
-
-exports.createPages = ({ graphql, actions }) => {
-    const { createPage } = actions;
-
-    return new Promise((resolve, reject) => {
-      graphql(`
-      {
-        allKenticoCloudItemArticle {
-          edges {
-            node {
-              fields {
-                slug
-              }
+const path = require(`path`) // highlight-line
+
+exports.onCreateNode = ({ node, actions: { createNodeField } }) => {
+  if (node.internal.type === `KenticoCloudItemArticle`) {
+    createNodeField({
+      node,
+      name: `slug`,
+      value: node.elements.url_pattern.value,
+    })
+  }
+}
+
+// highlight-start
+exports.createPages = async ({ graphql, actions }) => {
+  const { createPage } = actions
+
+  // Query data from Kentico
+  const result = await graphql(`
+    {
+      allKenticoCloudItemArticle {
+        edges {
+          node {
+            fields {
+              slug
             }
           }
         }
       }
-      `).then(result => {
-        result.data.allKenticoCloudItemArticle.edges.forEach(({ node }) => {
-            console.log(node.fields.slug);
-            createPage({
-                path: node.fields.slug,
-                component: path.resolve(`src/templates/article.js`),
-                context: {
-                    slug: node.fields.slug,
-                }
-            })
-        });
-        resolve();
-    });
-  });
+    }
+  `)
+
+  // Create pages
+  result.data.allKenticoCloudItemArticle.edges.forEach(({ node }) => {
+    console.log(node.fields.slug)
```

Comment:
should this console.log be here?

Reply:
I left whatever was here before 😄

Reply:
Legacy oversight! Would you mind removing as part of this PR, @browniebroke?